### ISSUE TITLE
Feature: Lighter and Darker colour shade shortcuts

### ISSUE THREAD

**Issue by therealrobster** [aseprite/aseprite#1360]
Describe your bug report or feature request here
Although you have a shading tool the issue with it is you have to create your own palette.  This is fine if you want to prep a palette and do that, but if I want to draw and shade I'd like to be able to press say L for lighter and the selected colour goes up one shade.  Press L again and it goes up another shade.  etc.  The same for the reverse. I could press D and it would go down a shade, keep pressing D and it would keep going down shades.  

They keys can be something else of course but it would allow me to shade SO much quicker than any other solution I can think of.  I hope you accept this feature request.

### Aseprite and System version

* Aseprite (version number, installer/portable/Steam/beta/dev):
* System (Win/OSX/Linux, version, distribution):

**Comment by dacap** [comment]
Hi @therealrobster, you already can do this. Try to configure keyboard shortcuts for left/right click trigger (e.g. `A` for left click, and `S` for right click):

![screen shot 2016-12-16 at 11 15 32 am](https://cloud.githubusercontent.com/assets/39654/21265707/64da0c1e-c381-11e6-9507-34fc7826f50f.png)

(I'm closing this issue but you can still make comments and I can re-open the issue if needed)

**Comment by therealrobster** [comment]
Hey there, thanks for the reply.  This wasn't quite what I meant.  Apologies if I wasn't clearer.

The idea is that I could with my left button brush be on say, a green colour.  If I press say L, then it would go slightly lighter shade of green.  If I press L again, it gets another shade lighter.  Each time I press L the shade goes one shade lighter on the selected colour.

The alternate is if I pressed say D, and the shade would go darker green.  D again, and another shade of green darker.

This way, I can be on ANY colour and just by pressing L and D (or whatever keys are chosen) I can very quickly shade items.

**Comment by therealrobster** [comment]
Just dropping in to see if you think this might end up coming to fruition one day?

**Comment by therealrobster** [comment]
This has become one of Krita's most loved tools.  Just checking in, any chance of seeing this in Aseprite?

**Comment by flamendless** [comment]
hmm are you saying that with the L and D keyboard shortcut, it will draw a light or dark shade without actually having a palette?

**Comment by therealrobster** [comment]
Sure, so if I for example press L and draw a line.  Then press L again, draw another line.  etc  It will keep drawing one shade lighter than the last time I pressed L.

The same for the opposite (D).  It will do a shade darker each time.  

This is all based on the colour I choose.  So I may select a nice mid range blue.  Each time I press L and draw it will get lighter and lighter.  Imagine how useful this is for shading work.   

Take this image for example (not mine)  https://streak.club/img/dXBsb2Fkcy9pbWFnZS81OTg2LnBuZw==/original/HGcl4W.png  You can just keep pressing D as you go to move down a shade level.

I would not recommend D and L as they're so far apart from each other.  They should be comfortable to have next to each other for moving up and down the shading levels.  I hope that made more sense...

**Comment by flamendless** [comment]
yes, that is indeed very useful. But, i have to wonder, how "high" or "low" would the next shade be from the selected base color?

**Comment by therealrobster** [comment]
Agreed. When I was writing it out and used that tree image as an example I was considering that.  I guess it could be an option.  How many steps between level changes.  More options = more complication but also allows for a more powerful tool.  I'm sure with testing a happy default could exist.  Then when the tool is selected, just like you have a brush size or brush type option at the top you could have a step count option for any tools that use this option.  I'd imagine you could use it for many.  Pencil, Paint Bucket, Line, Rectangle and Contour.

**Comment by flamendless** [comment]
hmmm indeed, that's a conceivable idea and feature request. Let's wait for the @dacap   reply about it

**Comment by dacap** [comment]
Hi! I'm thinking that this could be another ink (similar to the Shading ink), maybe a Hue/Saturation/Lightness ink, where you have three modes, and each mode move colors -step or +step on the selected channel (hue, saturation, or lightness) using left/right clicks (or keyboard shortcuts).

This is related to something I'm thinking in these days, something like a "super customization" possibility. For example: the possibility to trigger some complex logic (macro/script) with a key press, e.g. one key could start run a macro which consist in a serie of steps: 1) select this new ink mode, 2) press mouse button, 3) release the mouse button automatically, and then 4) restore the ink mode. (this could be a script or a macro in an extension #1403, related to #475 too)

**Comment by flamendless** [comment]
hmmm.. indeed. thats great @dacap

**Comment by therealrobster** [comment]
Just checking in to see if this got any love in any recent builds?

**Comment by yasmin-liban** [comment]
So I recently bought the software and was looking for this feature (it has on piskel witch was my alternative before I bought this), hope you can add this on a near future. Great tool I'm really enjoying it! @dacap
